File: edu/HRO/CMITDW Taal der Wiskunde/Lua/o72.lua
-- opdracht 7.2; puzzel 2
local strfind = string.find
print("i", "i^2", "i^3", "single")

for i = 1, 1000 do
	local square, cube = i^2, i^3
	local s = square..cube
	local hasNumbers = true
	
	for j = 0, 9 do
		-- check for each number in either strings
		if not strfind(s, j) then
			hasNumbers = false
			break
		end
	end
	
	if hasNumbers then
		local single = true
		
		for j = 0, 9 do
			local _, p2 = strfind(s, j)
			-- find any second instance of number
			if strfind(s, j, p2+1) then
				single = false
				break
			end
		end
		
		print(i, square, cube, single)
	end
end

File: HRO/CMITDW Taal der Wiskunde/o72.java
// opdracht 7.2; puzzel 2
public class o72
{
	public static void main(String[] args)
	{
		System.out.println("i\ti^2\ti^3\tsingle");
		
		for (int i=1; i<=1000; i++)
		{
			int square = (int) Math.pow(i, 2);
			int cube = (int) Math.pow(i, 3);
			String s = String.valueOf(square)+String.valueOf(cube);
			boolean hasNumbers = true;
			
			for (int j=0; j<=9; j++)
			{
				// check for each number in either strings
				if (!s.contains(String.valueOf(j)))
				{
					hasNumbers = false;
					break;
				}
			}
			
			if (hasNumbers)
			{
				boolean single = true;
				
				for (int j=0; j<=9; j++)
				{
					// ommitted first instance of number
					String sub = s.substring(s.indexOf(String.valueOf(j))+1);
					// find any second instance of number
					if (sub.contains(String.valueOf(j)))
					{
						single = false;
						break;
					}
				}
				
				System.out.println(i+"\t"+square+"\t"+cube+"\t"+single);
			}
		}
	}
}

